### PR TITLE
chore: Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Review by georglauterbach — DISMISSED** [review]
Thank you for this great addition - very much appreciated!

File: .github/workflows/default_on_push.yml
Comment (georglauterbach) [inline .github/workflows/default_on_push.yml:18]: We do not need the `write` permission here, right? We're pushing to GHCR, and I haven't read whether this requires some sort of extra-permission.

Comment (naveensrinivasan) [reply .github/workflows/default_on_push.yml:18]: Reading content is different from writing content. You probably need `packages:write` https://github.com/ossf/scorecard/blob/6ed6c9b70e837db3e94d91eb09c45fcba4bb6b15/.github/workflows/publishimage.yml#L20

But I would start with the least and then try it to before adding additional permissions.

**Review by wernerfred — CHANGES_REQUESTED** [review]
Thx for digging into this 👍🏻 
1 request for you: I would favor not to use inline comments. Write them in the line above pls.
1 request for @docker-mailserver/maintainers: I would favor to merge this after v11 got released. If we introduce a permission regression here (likely based on the comments) which interferes with v11 we will have to deal with 2 "major" issues

**Comment by naveensrinivasan** [comment]
> 1 request for you: I would favor not to use inline comments. Write them in the line above pls.

I have removed the inline comments.

**Review by polarathene — APPROVED** [review]
LGTM :+1: 

Cheers for the relevant links :)

---

> I would favor to merge this after v11 got released.

:100: :+1: 

---

Are wanting to address [`contributors`](https://github.com/docker-mailserver/docker-mailserver/blob/master/.github/workflows/contributors.yml) and [`docs-preview-deploy`](https://github.com/docker-mailserver/docker-mailserver/blob/master/.github/workflows/docs-preview-deploy.yml) workflows too?

Both of those look like they'd require multiple permission scopes. Would be nice if GH Actions was able to infer what was used/required with our current runs..

File: .github/workflows/default_on_push.yml
Comment (georglauterbach) [reply .github/workflows/default_on_push.yml:18]: Alright. Feature-freeze is over for `v11.0.0`. I'd like you to add the `packages:write` permission here :) Please also have a look at the comment by @polarathene (https://github.com/docker-mailserver/docker-mailserver/pull/2555#discussion_r853640460) so we can go ahead and merge this PR - it's a very nice addition! :D

**Review by wernerfred — APPROVED** [review]
LGTM

**Review by casperklein — APPROVED** [review]

**Comment by georglauterbach** [comment]
@naveensrinivasan please have a quick look at the left-over comments :)

File: .github/workflows/docs-production-deploy.yml
Comment (naveensrinivasan) [reply .github/workflows/docs-production-deploy.yml:74]: Took care of it. PTAL. Thanks

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan please have a quick look at the left-over comments :)

Took care of it. Thanks

**Comment by georglauterbach** [comment]
I added one line myself and updated the branch. @docker-mailserver/maintainers please have a look again :)

**Review by casperklein — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]
LGTM, thx

**Pull request merged**
